consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent2/ConsistentSamplingUtil.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler.consistent2;
+
+public final class ConsistentSamplingUtil {
+
+  private static final int RANDOM_VALUE_BITS = 56;
+  private static final long MAX_THRESHOLD = 1L << RANDOM_VALUE_BITS;
+  private static final long MAX_RANDOM_VALUE = MAX_THRESHOLD - 1;
+  private static final long INVALID_THRESHOLD = -1;
+  private static final long INVALID_RANDOM_VALUE = -1;
+
+  private ConsistentSamplingUtil() {}
+
+  /**
+   * Returns for a given threshold the corresponding sampling probability.
+   *
+   * <p>The returned value does not always exactly match the applied sampling probability, since
+   * some least significant binary digits may not be represented by double-precision floating point
+   * numbers.
+   *
+   * @param threshold the threshold
+   * @return the sampling probability
+   */
+  public static double calculateSamplingProbability(long threshold) {
+    checkThreshold(threshold);
+    return threshold * 0x1p-56;
+  }
+
+  /**
+   * Returns the closest sampling threshold that can be used to realize sampling with the given
+   * probability.
+   *
+   * @param samplingProbability the sampling probability
+   * @return the threshold
+   */
+  public static long calculateThreshold(double samplingProbability) {
+    checkProbability(samplingProbability);
+    return Math.round(samplingProbability * 0x1p56);
+  }
+
+  /**
+   * Calculates the adjusted count from a given threshold.
+   *
+   * <p>Returns 1, if the threshold is invalid.
+   *
+   * <p>Returns {@link Double#POSITIVE_INFINITY}, if the threshold is 0.
```

Comment:
threshold==0 is meant to indicate a non-probabilistic sampling decision. Should the method return 0 for this case?

Reply:
resolved by 9492eb4325483010af27f0301c0f1a5cfee4b805